src/DotLiquid.Tests/ExtendedFilterTests.cs

```diff
@@ -47,5 +44,109 @@ public void TestUpcaseFirst()
                 expected: "My great title",
                 template: "{{ 'my great title' | upcase_first }}");
         }
+
+        // Positive cases...
+        [TestCase("1582967411000", "2020-02-29T09:10:11+00:00")] // string
+        [TestCase(1582967411000d, "2020-02-29T09:10:11+00:00")] // double
+        [TestCase(1582967411000L, "2020-02-29T09:10:11+00:00")] // long
+        [TestCase(1582967411000UL, "2020-02-29T09:10:11+00:00")] // ulong
+        // Negative cases...
+        [TestCase(null, null)] // null safe
+        [TestCase("xxxxxxxx", "xxxxxxxx")] // ignore non-numeric string
+        [TestCase("1,582,967,411,000", "1,582,967,411,000")] // ignore string with thousand separator
+        [TestCase(0L, "1970-01-01T00:00:00+00:00")] // Epoch as long
+        [TestCase(1582967411000f, 1582967411000f)] // Ignore float as its precision of ~6-9 digits cannot accurately represent a timestamp in milliseconds
+        [TestCase(Int32.MaxValue, Int32.MaxValue)] // Ignore int, MaxValue ~= 1970-01-25T20:31:23+00:00
+        [TestCase(UInt32.MaxValue, UInt32.MaxValue)] // Ignore uint, MaxValue ~= 1970-02-19T17:02:47+00:00
+        public void TestUnixMs(object input, object expectedValue)
+        {
+            Helper.LockTemplateStaticVars(Template.NamingConvention, () =>
+            {
+                var actual = ExtendedFilters.UnixMs(context: _context, input: input);
+                Assert.AreEqual(
+                    expected: expectedValue,
+                    actual: actual is DateTimeOffset actualDateTimeOffset ? actualDateTimeOffset.ToString("yyyy-MM-ddTHH:mm:sszzz") : actual);
+            });
+        }
+
+        [Test]
+        public void TestTimeZone()
+        {
+            Context context = new Context(CultureInfo.CurrentCulture);
+
+            // Test UTC specified DateTime
+            var utcDateTime = new DateTime(2020, 2, 29, 12, 10, 11, DateTimeKind.Utc);
+            Assert.AreEqual("2020-02-29T12:10:11+00:00", ToIso8601DateString(ExtendedFilters.TimeZone(context: context, input: utcDateTime, convertToTimezoneId: "UTC")));
+            Assert.AreEqual("2020-02-29T12:10:11+00:00", ToIso8601DateString(ExtendedFilters.TimeZone(context: context, input: utcDateTime, convertToTimezoneId: "Local")));
```

Comment:
How will this pass on developer machines?

Reply:
I've removed Local, thinking again I can't see an obvious use case for it.